apps/openmw/mwgui/alchemywindow.cpp

```diff
@@ -250,4 +279,63 @@ namespace MWGui
 
         update();
     }
+
+    void AlchemyWindow::addRepeatController(MyGUI::Widget *widget)
+    {
+        MyGUI::ControllerItem* item = MyGUI::ControllerManager::getInstance().createItem(Controllers::ControllerRepeatEvent::getClassTypeName());
+        Controllers::ControllerRepeatEvent* controller = item->castType<Controllers::ControllerRepeatEvent>();
+        controller->eventRepeatClick += MyGUI::newDelegate(this, &AlchemyWindow::onRepeatClick);
+        controller->setRepeat(sCountChangeInitialPause, sCountChangeInterval);
+        MyGUI::ControllerManager::getInstance().addItem(widget, controller);
+    }
+
+    void AlchemyWindow::onIncreaseButtonPressed(MyGUI::Widget* _sender, int _left, int _top, MyGUI::MouseButton _id)
+    {
+        addRepeatController(_sender);
+        onIncreaseButtonTriggered();
+    }
+
+    void AlchemyWindow::onDecreaseButtonPressed(MyGUI::Widget* _sender, int _left, int _top, MyGUI::MouseButton _id)
+    {
+        addRepeatController(_sender);
+        onDecreaseButtonTriggered();
+    }
+
+    void AlchemyWindow::onRepeatClick(MyGUI::Widget* widget, MyGUI::ControllerItem* controller)
+    {
+        if (widget == mIncreaseButton)
+            onIncreaseButtonTriggered();
+        else if (widget == mDecreaseButton)
+            onDecreaseButtonTriggered();
+    }
+
+    void AlchemyWindow::onCountButtonReleased(MyGUI::Widget *_sender, int _left, int _top, MyGUI::MouseButton _id)
+    {
+        MyGUI::ControllerManager::getInstance().removeItem(_sender);
+    }
+
+    void AlchemyWindow::onCountValueChanged(int value)
+    {
+        mBrewCountEdit->setValue(std::abs(value));
+    }
+
+    void AlchemyWindow::onIncreaseButtonTriggered()
+    {
+        int currentCount = mBrewCountEdit->getValue();
+
+        // prevent overflows, and prevent entering INT_MIN since abs(INT_MIN) is undefined
+        if (currentCount == INT_MAX || currentCount == INT_MIN+1)
```

Comment:
This is the C way of doing it. Let's go for the C++ version (std::numeric_limits).

Reply:
I took this code from the trading window. Change it thre too?